src/utils/ThirdPartyApi.js

```diff
@@ -0,0 +1,63 @@
+const API_KEY = "dd695f0f3720453e9025323b3c051bb8";
+const BASE_URL =
+  process.env.NODE_ENV === "production"
+    ? "https://nomoreparties.co/news/v2"  // Proxy URL for production
+    : "https://newsapi.org/v2";  // Also using proxy URL for development
+
+export const handleServerResponse = (res) => {
+  return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+};
+
+// Function to fetch news articles based on search query
+export const getNews = async (query) => {
+  const endpoint = `${BASE_URL}/everything?q=${encodeURIComponent(
+    query
+  )}&apiKey=${API_KEY}&pageSize=100&from=${getFormattedDate(
+    7
+  )}&to=${getFormattedDate(0)}`;
+
+  try {
+    const response = await fetch(endpoint);
+    if (!response.ok) {
+      throw new Error("Failed to fetch news articles");
+    }
+
+    const data = await response.json();
+
+    console.log("Fetched articles from API:", data.articles);
+
+    // Map through the articles to extract the necessary fields
+    const articles = data.articles.map((article) => ({
+      sourceName: article.source.name,
+      title: article.title,
+      publishedAt: formatDate(article.publishedAt),
```

Comment:
`      publishedAt: formatDate(article.publishedAt),
`

I think this is not needed here. Please, format the date only when you display it on the page, otherwise it's formatted here and you send this date to the server which doesn't recognize it (if it's not in English)